src/App.tsx

```diff
@@ -1,26 +1,278 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from './types/Todo';
+import * as helpers from './api/todos';
+import { Status } from './types/status';
+import { Emessage } from './types/Emessage';
+import { Header } from './Header';
+import { TodoList } from './TodoList';
+import { Footer } from './Footer';
+import { Notification } from './Notification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [queryStatus, setQueryStatus] = useState(Status.all);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errMessage, setErrMessage] = useState(Emessage.null);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inputText, setInputText] = useState('');
```

Comment:
Make this state local for a header as well and pass it as a param to the addTodo function so you can pass fewer unnecessary props